langchain4j-oracle/src/main/java/dev/langchain4j/store/embedding/oracle/OracleEmbeddingStore.java

```diff
@@ -97,41 +97,31 @@ private OracleEmbeddingStore(Builder builder) {
     }
 
     /**
-     * Creates an index on the {@link EmbeddingTable#embeddingColumn()}, if configured to do so by
-     * {@link Builder#vectorIndex(CreateOption)}.
-     *
-     * @param builder Builder that configures an embedding store. Not null.
-     *
-     * @throws SQLException If a database error prevents the index from being created.
+     * Creates all index described by the builder.
+     * @param builder Then builder.
+     * @throws SQLException If an error occurs during index creation.
      */
     private static void createIndex(Builder builder) throws SQLException {
 
-        if (builder.vectorIndexCreateOption == CreateOption.DO_NOT_CREATE)
-            return;
-
         try (Connection connection = builder.dataSource.getConnection();
```

Comment:
We should not acquire a Connection unless we know we gonna use it. We should check if any index creation is actually required first.